File: shoring/compat/src/test/java/com/panopset/compat/test/TextScramblerTest.java
package com.panopset.compat.test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;
import org.junit.jupiter.api.Test;

import com.panopset.compat.Randomop;
import com.panopset.compat.Stringop;
import com.panopset.compat.TextScrambler;

public class TextScramblerTest {

  private static final String TEST_PWD = "4traN";
  private static final String TEN_MILES = "ten miles";
  
  private static final String MANY_HAPPY_RETURNS = "FRqy4hq6gPqAWnN-D-gQ3wQG"
      + Stringop.DOS_RTN + "pPsEzqGA7yu7F8sczbGBMM8_wmeBZg==\n" + Stringop.DOS_RTN;
  private final String txtLong = "This is not very long but long enough";
  private String txtShort = "FOO";
  private String txtRandom = "" + Randomop.nextLong();

  @Test
  void testScrambler() throws Exception {
    ts = new TextScrambler();
    testScramblerOn(txtShort);
    testScramblerOn(txtLong);
    testScramblerOn(txtRandom);
    testScramblerOn(TEN_MILES);
    assertEquals(TEN_MILES, ts.decrypt(TEST_PWD, MANY_HAPPY_RETURNS));
  }

  private void testScramblerOn(String txt0) throws Exception {
    String txt1 = txt0 + "a";
    String scrt0 = ts.encrypt(TEST_PWD, txt0);
    String scrt1 = ts.encrypt(TEST_PWD, txt1);
    assertFalse(scrt0.equals(txt0));
    assertFalse(scrt1.equals(txt1));
    assertFalse(scrt0.substring(0, 20).equals(scrt1.substring(0, 20)));
    String rslt0 = ts.decrypt(TEST_PWD, scrt0);
    String rslt1 = ts.decrypt(TEST_PWD, scrt1);
    assertEquals(rslt0, txt0);
    assertEquals(rslt1, txt1);

  }

  TextScrambler ts;
}

File: code/shoring/compat/src/test/kotlin/com/panopset/compat/test/TextScramblerTest.kt
package com.panopset.compat.test

import com.panopset.compat.Randomop
import com.panopset.compat.Stringop
import com.panopset.compat.TextScrambler
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class TextScramblerTest {
    private val txtLong = "This is not very long but long enough"
    private val txtShort = "FOO"
    private val txtRandom = "" + Randomop.nextLong()

    @Test
    fun testScrambler() {
        val ts = TextScrambler()
        testScramblerOn(ts, txtShort)
        testScramblerOn(ts, txtLong)
        testScramblerOn(ts, txtRandom)
        testScramblerOn(ts, TEN_MILES)
        Assertions.assertEquals(TEN_MILES, ts.decrypt(TEST_PWD, MANY_HAPPY_RETURNS))
    }

    private fun testScramblerOn(ts: TextScrambler, txt0: String) {
        val txt1 = txt0 + "a"
        val scrt0 = ts.encrypt(TEST_PWD, txt0)
        val scrt1 = ts.encrypt(TEST_PWD, txt1)
        Assertions.assertFalse(scrt0 == txt0)
        Assertions.assertFalse(scrt1 == txt1)
        Assertions.assertFalse(scrt0.substring(0, 20) == scrt1.substring(0, 20))
        val rslt0 = ts.decrypt(TEST_PWD, scrt0)
        val rslt1 = ts.decrypt(TEST_PWD, scrt1)
        Assertions.assertEquals(rslt0, txt0)
        Assertions.assertEquals(rslt1, txt1)
    }

    companion object {
        private const val TEST_PWD = "4traN"
        private const val TEN_MILES = "ten miles"
        private const val MANY_HAPPY_RETURNS = ("FRqy4hq6gPqAWnN-D-gQ3wQG"
                + Stringop.DOS_RTN + "pPsEzqGA7yu7F8sczbGBMM8_wmeBZg==\n" + Stringop.DOS_RTN)
    }
}